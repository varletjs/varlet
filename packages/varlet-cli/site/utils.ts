import { get } from 'lodash-es'
import { StyleProvider, type Theme, withSiteConfigNamespace } from '@varlet/cli/client'

export interface Menu {
  doc: string
  text: Record<string, string>
  type: MenuTypes
  useMobile?: boolean
}

export enum MenuTypes {
  TITLE = 1,
  COMPONENT = 2,
  DOCUMENTATION = 3,
}

export function removeEmpty(object: Record<string, string> = {}) {
  return Object.keys(object).reduce((record: Record<string, string>, key) => {
    const value = object[key]
    value && (record[key] = value)
    return record
  }, {})
}

export function isPhone() {
  return /Android|webOS|iPhone|iPod|BlackBerry|Pad/i.test(navigator.userAgent)
}

export function inIframe() {
  return window.self !== window.top
}

export function setTheme(config: Record<string, any>, name: Theme) {
  const styleVars = withSiteConfigNamespace(get(config, name, {}))

  StyleProvider(styleVars)
}

export function utoa(data: string): string {
  return btoa(unescape(encodeURIComponent(data)))
}
