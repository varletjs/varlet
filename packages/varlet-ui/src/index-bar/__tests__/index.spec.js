import IndexBar from '..'
import IndexAnchor from '../../index-anchor'
import VarIndexBar from '../IndexBar'
import VarIndexAnchor from '../../index-anchor/IndexAnchor'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'
import { delay, mockScrollTo } from '../../utils/jest'

mockScrollTo(HTMLElement)

function mockIndexBarOwnTop() {
  const originForEach = Array.prototype.forEach

  Array.prototype.forEach = function (fn, thisArg) {
    let changedArr = this

    if (this && this.map) {
      changedArr = this.map((value, index) => {
        if (value.ownTop && !value.ownTop.value) value.ownTop.value = index % 2 === 0 ? index * 10 : index * -10

        return value
      })
    }

    originForEach.call(changedArr, fn, thisArg)
  }

  return {
    mockRestore() {
      Array.prototype.forEach = originForEach
    },
  }
}

const Wrapper = {
  components: {
    [VarIndexBar.name]: VarIndexBar,
    [VarIndexAnchor.name]: VarIndexAnchor,
  },
  template: `
    <var-index-bar>
    <var-index-anchor index="A">test A</var-index-anchor>
    <var-index-anchor index="B">test B</var-index-anchor>
    </var-index-bar>
  `,
}

const clickHandle = jest.fn()
const changeHandle = jest.fn()
const Wrapper2 = {
  template: `
    <div style="height: 50px; overflow: auto">
      <var-index-bar @click="clickHandle" @change="changeHandle" ref="bar" highlight-color="purple">
        <var-index-anchor index="A">test A</var-index-anchor>
        <p>test</p>
        <p>test</p>
        <p>test</p>
        <var-index-anchor index="B">test B</var-index-anchor>
        <p>test</p>
        <p>test</p>
        <p>test</p>
        <var-index-anchor index="C">test C</var-index-anchor>
        <p>test</p>
        <p>test</p>
        <p>test</p>
        <var-index-anchor index="D">test D</var-index-anchor>
        <p>test</p>
        <p>test</p>
        <p>test</p>
        <var-index-anchor index="E">test E</var-index-anchor>
      </var-index-bar>
    </div>
  `,
  components: {
    [VarIndexBar.name]: VarIndexBar,
    [VarIndexAnchor.name]: VarIndexAnchor,
  },
  methods: {
    clickHandle,
    changeHandle,
  },
}

test('test indexBar and indexAnchor use', () => {
  const app = createApp({}).use(IndexBar).use(IndexAnchor)

  expect(app.component(IndexBar.name)).toBeTruthy()
  expect(app.component(IndexAnchor.name)).toBeTruthy()
})

describe('test index-bar component props', () => {
  test('test sticky prop and it is value equal false', async () => {
    const wrapper = mount(Wrapper, {
      props: {
        sticky: false,
      },
      attachTo: document.body,
    })

    await delay(100)
    expect(wrapper.find('.var-sticky').exists()).toBeFalsy()

    wrapper.unmount()
  })

  test('test index-bar stickyOffsetTop', async () => {
    const wrapper = mount(Wrapper, {
      props: {
        stickyOffsetTop: '36px',
      },
      attachTo: document.body,
    })

    await wrapper.trigger('scroll')
    await delay(100)

    expect(wrapper.find('.var-sticky__wrapper').attributes('style')).toContain('top: 36px;')

    wrapper.unmount()
  })

  test('test index-bar hideList', async () => {
    const wrapper = mount(Wrapper, {
      props: {
        hideList: true,
      },
      attachTo: document.body,
    })

    await delay(100)
    expect(wrapper.find('.var-index-bar__anchor-list').attributes('style')).toContain('display: none;')

    wrapper.unmount()
  })

  test('test index-bar zIndex', async () => {
    const wrapper = mount(Wrapper, {
      props: {
        zIndex: 2,
      },
      attachTo: document.body,
    })

    await delay(100)
    expect(wrapper.find('.var-sticky').attributes('style')).toContain('z-index: 2;')

    wrapper.unmount()
  })

  test('test index-bar highlightColor', async () => {
    const wrapper = mount(Wrapper2, { attachTo: document.body })

    await delay(100)
    await wrapper.find('.var-index-bar__anchor-item').trigger('click')

    expect(wrapper.find('.var-index-bar__anchor-item--active').attributes('style')).toContain('color: purple')

    wrapper.unmount()
  })
})

describe('test index-bar events', () => {
  test('test index-bar click event', async () => {
    jest.clearAllMocks()

    const wrapper = mount(Wrapper2, { attachTo: document.body })

    await delay(100)

    const anchorItems = wrapper.findAll('.var-index-bar__anchor-item')

    await anchorItems[0].trigger('click')
    await delay(100)

    expect(anchorItems[0].classes()).toContain('var-index-bar__anchor-item--active')

    expect(clickHandle).toHaveBeenCalledTimes(1)

    wrapper.unmount()
  })

  test('test indexBar scrollTo method', async () => {
    const wrapper = mount(Wrapper2, { attachTo: document.body })

    await delay(100)

    wrapper.vm.$refs.bar.scrollTo('A')

    await delay(100)

    expect(wrapper.find('.var-index-bar__anchor-item--active').text()).toBe('A')

    wrapper.unmount()
  })

  test('test indexBar scroll to trigger change event', async () => {
    jest.clearAllMocks()

    const { mockRestore } = mockIndexBarOwnTop()
    const wrapper = mount(Wrapper2, { attachTo: document.body })

    await delay(100)

    wrapper.element.scrollTop = 150
    await wrapper.trigger('scroll')

    expect(changeHandle).toHaveBeenCalled()

    mockRestore()
    wrapper.unmount()
  })
})
