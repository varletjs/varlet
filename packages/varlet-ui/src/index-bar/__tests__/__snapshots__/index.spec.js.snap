// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test click anchor to trigger change event 1`] = `
"<div style=\\"height: 50px; overflow: auto;\\">
  <div class=\\"var-index-bar\\">
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 0px;\\">
        <div class=\\"var-index-anchor\\">test A</div>
      </div>
    </div>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 0px;\\">
        <div class=\\"var-index-anchor\\">test B</div>
      </div>
    </div>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 0px;\\">
        <div class=\\"var-index-anchor\\">test C</div>
      </div>
    </div>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 0px;\\">
        <div class=\\"var-index-anchor\\">test D</div>
      </div>
    </div>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <p>test</p>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 0px;\\">
        <div class=\\"var-index-anchor\\">test E</div>
      </div>
    </div>
    <ul class=\\"var-index-bar__anchor-list\\" style=\\"z-index: 3; display: block;\\">
      <li class=\\"var-index-bar__anchor-item\\" style=\\"\\">A</li>
      <li class=\\"var-index-bar__anchor-item var-index-bar__anchor-item--active\\" style=\\"color: purple;\\">B</li>
      <li class=\\"var-index-bar__anchor-item\\">C</li>
      <li class=\\"var-index-bar__anchor-item\\">D</li>
      <li class=\\"var-index-bar__anchor-item\\">E</li>
    </ul>
  </div>
</div>"
`;

exports[`test indexBar example 1`] = `
"<div class=\\"var-index-bar\\">
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 A</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">A 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">A 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 B</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">B 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">B 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 C</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">C 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">C 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 D</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">D 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">D 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 E</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">E 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">E 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 F</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">F 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">F 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 G</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">G 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">G 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 H</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">H 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">H 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 I</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">I 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">I 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 J</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">J 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">J 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 K</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">K 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">K 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 L</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">L 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">L 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 M</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">M 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">M 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 N</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">N 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">N 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 O</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">O 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">O 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 P</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">P 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">P 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 Q</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Q 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Q 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 R</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">R 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">R 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 S</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">S 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">S 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 T</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">T 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">T 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 U</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">U 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">U 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 V</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">V 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">V 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 W</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">W 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">W 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 X</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">X 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">X 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 Y</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Y 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Y 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <div>
    <div class=\\"var-sticky\\" style=\\"z-index: 1; width: 0px; height: 0px;\\">
      <div class=\\"var-sticky__wrapper\\" style=\\"z-index: 1; position: fixed; width: 0px; height: 0px; left: 0px; top: 50px;\\">
        <div class=\\"var-index-anchor var-index-anchor__example\\">标题 Z</div>
      </div>
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Z 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div class=\\"var-cell\\">
      <!--v-if-->
      <div class=\\"var-cell__content\\">
        <div class=\\"var-cell__title\\">Z 文本</div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <ul class=\\"var-index-bar__anchor-list\\" style=\\"z-index: 3; display: block;\\">
    <li class=\\"var-index-bar__anchor-item\\">A</li>
    <li class=\\"var-index-bar__anchor-item\\">B</li>
    <li class=\\"var-index-bar__anchor-item\\">C</li>
    <li class=\\"var-index-bar__anchor-item\\">D</li>
    <li class=\\"var-index-bar__anchor-item\\">E</li>
    <li class=\\"var-index-bar__anchor-item\\">F</li>
    <li class=\\"var-index-bar__anchor-item\\">G</li>
    <li class=\\"var-index-bar__anchor-item\\">H</li>
    <li class=\\"var-index-bar__anchor-item\\">I</li>
    <li class=\\"var-index-bar__anchor-item\\">J</li>
    <li class=\\"var-index-bar__anchor-item\\">K</li>
    <li class=\\"var-index-bar__anchor-item\\">L</li>
    <li class=\\"var-index-bar__anchor-item\\">M</li>
    <li class=\\"var-index-bar__anchor-item\\">N</li>
    <li class=\\"var-index-bar__anchor-item\\">O</li>
    <li class=\\"var-index-bar__anchor-item\\">P</li>
    <li class=\\"var-index-bar__anchor-item\\">Q</li>
    <li class=\\"var-index-bar__anchor-item\\">R</li>
    <li class=\\"var-index-bar__anchor-item\\">S</li>
    <li class=\\"var-index-bar__anchor-item\\">T</li>
    <li class=\\"var-index-bar__anchor-item\\">U</li>
    <li class=\\"var-index-bar__anchor-item\\">V</li>
    <li class=\\"var-index-bar__anchor-item\\">W</li>
    <li class=\\"var-index-bar__anchor-item\\">X</li>
    <li class=\\"var-index-bar__anchor-item\\">Y</li>
    <li class=\\"var-index-bar__anchor-item\\">Z</li>
  </ul>
</div>"
`;
